src/sessions/SessionManager.ts

```diff
@@ -125,6 +132,19 @@ export class SessionManager {
         return this.attributes;
     }
 
+    /**
+     * Adds custom session attributes to the session's attributes
+     * @param customAttributesMap object containing custom attribute data in the form of key, value pairs
+     */
+    public setCustomAttributes(customAttributesMap: {
+        [k: string]: string | number | boolean;
+    }) {
+        const attributeKeys = Object.keys(customAttributesMap);
+        for (const attribute of attributeKeys) {
+            this.attributes[attribute] = customAttributesMap[attribute];
+        }
```

Comment:
suggestion: The [spread operator ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) can help here. Note that the second spread operator (`...customAttributeMap`) takes precedence, so attributes will be overridden with new attributes in this example.
```suggestion
        this.attributes = { ...this.attributes, ...customAttributeMap };
```

Reply:
Updated to use suggestion, thanks!